pkg/controllers/provisioning/controller.go

```diff
@@ -71,6 +73,19 @@ func (c *Controller) Builder(_ context.Context, m manager.Manager) operatorcontr
 	return operatorcontroller.Adapt(controllerruntime.
 		NewControllerManagedBy(m).
 		For(&v1.Pod{}).
+		// Only enqueue pods that have the node name not set. We do this here so that we can enqueue the pod
+		// requests from nodes that are deleting
+		WithEventFilter(predicate.NewPredicateFuncs(func(o client.Object) bool {
```

Comment:
Should we have changed L61? I don't see how this triggers reconciliation for tainted nodes now?

Reply:
Yes good catch!